x-pack/plugin/core/src/main/java/org/elasticsearch/protocol/xpack/XPackInfoResponse.java

```diff
@@ -203,11 +205,28 @@ public int hashCode() {
 
         @Override
         public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-            builder.startObject()
-                .field("uid", uid)
-                .field("type", type)
-                .field("mode", mode)
-                .field("status", status.label());
+            builder.startObject();
+            builder.field("uid", uid);
+
+            final boolean acceptEnterprise = params.paramAsBoolean("accept_enterprise", false);
```

Comment:
I don't think it is possible to just fetch `accept_enterprise` parameter here? Unlike `RestGetLicenseAction`, the parameter is not passed into the `toXContent` method. Since `XPackInfoRequest` no longer takes a version parameter, I think we'll need to do something similar to `RestGetLicenseAction` for `RestXPackInfoAction`, i.e. passing `hide_enterprise` from the rest handler into `toXContent` method so it is available in this method.

Reply:
We chatted and it made me realise that all rest request parameters are automatically made available to the `toXContent` method by `RestToXContentListener`. So there is no issue here.